app/components/edit/EditServiceChildCollection.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, useState } from 'react';
+
+/**
+ * Renders a child property collection of a Service as a list of individual components,
+ * called ResourceObjectItem. Enables the user to add, modify, and delete members of
+ * the collection and save to the DB.
+ *
+ * In the future, this component could possibly be abstracted to replace the
+ * EditCollection HOC class-based JSX component that does essentially the same thing
+ * for child property collections of Resources.
+ */
+interface ComponentProps {
+  index: number;
+  item: any;
+  handleItemChange: any;
+}
+
+interface ResourceItem {
+  keyId?: number;
+  id?: string | number;
+}
+
+export const EditServiceChildCollection = <T extends ResourceItem> ({
+  initialCollectionData,
+  handleCollectionChange,
+  ResourceObjectItem,
+  label,
+  blankTemplateObj,
+  buttonText,
+  propertyKeyName,
+}: {
+  initialCollectionData: any[];
+  handleCollectionChange: (field: string, value: T[]) => void;
+  ResourceObjectItem: FC<ComponentProps>;
+  label: string;
+  blankTemplateObj: T;
+  buttonText: string;
+  propertyKeyName: string;
+}) => {
+  const [resourceCollection, setResourceCollection] = useState<T[]>(
+    initialCollectionData || [],
+  );
+
+  const addItem = () => {
+    const newItem = {
+      ...blankTemplateObj,
+      keyId: Math.random(),
+    };
+
+    setResourceCollection([...resourceCollection, newItem]);
+  };
+
+  const handleItemChange = (index: number, item: any) => {
+    const newCollection: T[] = [
+      ...resourceCollection.slice(0, index),
+      item,
+      ...resourceCollection.slice(index + 1),
+    ];
+
+    setResourceCollection(newCollection);
+    handleCollectionChange(propertyKeyName, newCollection);
+  };
+
+  const removeItem = (index: number) => {
+    const newCollection = [
+      ...resourceCollection.slice(0, index),
+      ...resourceCollection.slice(index + 1),
+    ];
+
+    setResourceCollection(newCollection);
+    handleCollectionChange(propertyKeyName, newCollection);
+  };
+
+  const createItemComponents = (itemComponentCollection: T[]) => (
+    itemComponentCollection.map((item, index) => (
```

Comment:
Does this need to be a function? You only ever call this a single time per render, so I think you can just inline it. For example:

```tsx
const itemComponents = resourceCollection.map((item, index) => (
  // Everything else that you have here in the function
));

return (
   <>
      <label htmlFor="edit-item">{label}</label>
      <div className="edit--section--list--item--sublist">
        {itemComponents}
      </div>
```

Reply:
Oh yes, good catch. No need for that. Thank you.